src/Traits/NovaTestTrait.php

```diff
@@ -2,14 +2,108 @@
 
 namespace RonasIT\Support\Traits;
 
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Foundation\Testing\TestCase;
+use Illuminate\Support\Str;
+use Illuminate\Testing\TestResponse;
+
 trait NovaTestTrait
 {
-    public function novaSearchParams(array $filters, string $search = '', int $perPage = 25): array
+    protected function novaSearchParams(array $filters, string $search = '', int $perPage = 25): array
     {
         return [
             'search' => $search,
             'filters' => base64_encode(json_encode($filters)),
             'perPage' => $perPage,
         ];
     }
+
+    protected function novaCreateResourceAPICall(string $resourceClass, ?array $data = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->postJson($resourceUri, $data);
+    }
+
+    protected function novaUpdateResourceAPICall(string $resourceClass, int $resourceId, ?array $data = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->putJson("{$resourceUri}/{$resourceId}", $data);
+    }
+
+    protected function novaGetResourceAPICall(string $resourceClass, int $resourceId, ?array $data = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->getJson("{$resourceUri}/{$resourceId}", $data);
+    }
+
+    protected function novaSearchResourceAPICall(string $resourceClass, ?array $request = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->json('get', $resourceUri, $request);
+    }
+
+    protected function novaGetCreationFieldsAPICall(string $resourceClass): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->getJson("{$resourceUri}/creation-fields");
+    }
+
+    protected function novaRunActionAPICall(string $resourceClass, string $actionClass, ?array $request = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        $actionUri = app($actionClass)->uriKey();
+
+        return $this->json('POST', "{$resourceUri}/action?action={$actionUri}", $request);
```

Comment:
```suggestion
        return $this->postJson("{$resourceUri}/action?action={$actionUri}", $request);
```

Reply:
As strange as it may sound, tests won't work like that, I checked